### DISCUSSION TITLE
KDL query implementation

### DISCUSSION THREAD

**Discussion by Patitotective** [kdl-org/kdl#302] [Q&A]
I'm trying to implement the KDL query language (for my [kdl-nim](https://github.com/Patitotective/kdl-nim)) but it's hard to create it from scratch because I've never implemented a query language before and I wanted to know if there was any already existing implementation of the KDL query language, if so please let me know, thanks.

**Comment by larsgw** [comment]
I [implemented it in `kdljs`](https://github.com/kdl-org/kdljs/blob/main/src/queryEngine.js), though I think the spec might have changed since then (like removing the mapping syntax). I also never put it in the readme apparently, oops.

Edit: I didn't add support for sibling selectors so that might mean the structure of this implementation is unoptimal, but I hope it still helps.

**Reply by Patitotective** [reply]
Thank you.

**Comment by Patitotective** [comment]
It would also be cool if KQL had a grammar in `QUERY-SPEC.md`, like `SPEC.md`.
